typed-arrow-dyn/tests/views.rs

```diff
@@ -760,6 +764,144 @@ fn projected_views() -> Result<(), DynViewError> {
     Ok(())
 }
 
+#[test]
+fn nested_projection_masks_drive_dyn_projection() -> Result<(), DynViewError> {
```

Comment:
The new integration test only covers struct/list projections driven by a mask. There is no regression test exercising map projections, union projections, or the error cases (invalid paths, missing children, reordered map entries). Given how much logic was added in `projection.rs` and `view.rs`, a couple of targeted tests around those scenarios would make the change far “leaner” and protect against the bugs above.